lldb-eval/eval_test.cc

```diff
@@ -1347,6 +1347,21 @@ TEST_F(EvalTest, TestValueScope) {
   EXPECT_THAT(Scope("var").Eval("z_"),
               IsError("use of undeclared identifier 'z_'"));
 
+  // In "value" scope `this` refers to the scope object.
+  EXPECT_THAT(Scope("var").Eval("this->y_"), IsEqual("2.5"));
+  EXPECT_THAT(Scope("var").Eval("(*this).y_"), IsEqual("2.5"));
+
+  // Test for the "artificial" value, i.e. created by the expression.
+  lldb::SBError error;
+  lldb::SBValue scope_var =
+      lldb_eval::EvaluateExpression(frame_, "(test_scope::Value&)bytes", error);
+  EXPECT_TRUE(scope_var.IsValid());
+  EXPECT_TRUE(error.Success());
+
+  EvaluatorHelper scope(scope_var, true);
+  EXPECT_THAT(scope.Eval("this->y_"), IsEqual("2.5"));
+  EXPECT_THAT(scope.Eval("(*this).y_"), IsEqual("2.5"));
```

Comment:
Maybe add a test to make sure that `this` is undeclared when evaluated in the frame context?

Reply:
That's a good idea, done!